### ISSUE TITLE
puppetserver_metrics_ensure => absent is absent minded

### ISSUE THREAD

**Issue by tkishel** [puppetlabs/puppetlabs-puppet_metrics_collector#16]
AKA: there is no clean uninstall

```
[root@pe-201901-master ~]# puppet apply -e "class { 'puppet_metrics_collector': }" --modulepath=/root/puppet_modules
Notice: Compiled catalog for pe-201901-master.puppetdebug.vlan in environment production in 0.14 seconds
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/Cron[puppetserver_metrics_collection]/ensure: created
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/Cron[puppetserver_metrics_tidy]/ensure: created
Notice: /Stage[main]/Puppet_metrics_collector::Puppetdb/Puppet_metrics_collector::Pe_metric[puppetdb]/Cron[puppetdb_metrics_collection]/ensure: created
Notice: /Stage[main]/Puppet_metrics_collector::Puppetdb/Puppet_metrics_collector::Pe_metric[puppetdb]/Cron[puppetdb_metrics_tidy]/ensure: created
Notice: /Stage[main]/Puppet_metrics_collector::Orchestrator/Puppet_metrics_collector::Pe_metric[orchestrator]/Cron[orchestrator_metrics_collection]/ensure: created
Notice: /Stage[main]/Puppet_metrics_collector::Orchestrator/Puppet_metrics_collector::Pe_metric[orchestrator]/Cron[orchestrator_metrics_tidy]/ensure: created
Notice: Applied catalog in 0.23 seconds
```

```
[root@pe-201901-master ~]# crontab -l
# HEADER: This file was autogenerated at 2019-03-19 22:55:24 +0000 by puppet.
# HEADER: While it can still be managed manually, it is definitely not recommended.
# HEADER: Note particularly that the comments starting with 'Puppet Name' should
# HEADER: not be deleted, as doing so could cause duplicate cron jobs.
# Puppet Name: puppetserver_metrics_collection
*/5 * * * * /opt/puppetlabs/puppet-metrics-collector/scripts/tk_metrics --metrics_type puppetserver --output-dir /opt/puppetlabs/puppet-metrics-collector/puppetserver --no-print
# Puppet Name: puppetserver_metrics_tidy
20 1 * * * /opt/puppetlabs/puppet-metrics-collector/scripts/puppetserver_metrics_tidy
# Puppet Name: puppetdb_metrics_collection
*/5 * * * * /opt/puppetlabs/puppet-metrics-collector/scripts/tk_metrics --metrics_type puppetdb --output-dir /opt/puppetlabs/puppet-metrics-collector/puppetdb --no-print
# Puppet Name: puppetdb_metrics_tidy
30 0 * * * /opt/puppetlabs/puppet-metrics-collector/scripts/puppetdb_metrics_tidy
# Puppet Name: orchestrator_metrics_collection
*/5 * * * * /opt/puppetlabs/puppet-metrics-collector/scripts/tk_metrics --metrics_type orchestrator --output-dir /opt/puppetlabs/puppet-metrics-collector/orchestrator --no-print
# Puppet Name: orchestrator_metrics_tidy
0 0 * * * /opt/puppetlabs/puppet-metrics-collector/scripts/orchestrator_metrics_tidy
```

```
[root@pe-201901-master ~]# puppet apply -e "class { 'puppet_metrics_collector': puppetserver_metrics_ensure => absent }" --modulepath=/root/puppet_modules
Notice: Compiled catalog for pe-201901-master.puppetdebug.vlan in environment production in 0.17 seconds
Warning: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/File[/opt/puppetlabs/puppet-metrics-collector/puppetserver]: Could not back up file of type directory
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/File[/opt/puppetlabs/puppet-metrics-collector/puppetserver]: Not removing directory; use 'force' to override
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/File[/opt/puppetlabs/puppet-metrics-collector/puppetserver]/ensure: removed
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/File[/opt/puppetlabs/puppet-metrics-collector/scripts/puppetserver_config.yaml]/ensure: removed
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/Cron[puppetserver_metrics_collection]/ensure: removed
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/File[/opt/puppetlabs/puppet-metrics-collector/scripts/puppetserver_metrics_tidy]/ensure: removed
Notice: /Stage[main]/Puppet_metrics_collector::Puppetserver/Puppet_metrics_collector::Pe_metric[puppetserver]/Cron[puppetserver_metrics_tidy]/ensure: removed
Notice: Applied catalog in 0.24 seconds
```

```
[root@pe-201901-master ~]# crontab -l
# HEADER: This file was autogenerated at 2019-03-19 22:55:39 +0000 by puppet.
# HEADER: While it can still be managed manually, it is definitely not recommended.
# HEADER: Note particularly that the comments starting with 'Puppet Name' should
# HEADER: not be deleted, as doing so could cause duplicate cron jobs.
# Puppet Name: puppetdb_metrics_collection
*/5 * * * * /opt/puppetlabs/puppet-metrics-collector/scripts/tk_metrics --metrics_type puppetdb --output-dir /opt/puppetlabs/puppet-metrics-collector/puppetdb --no-print
# Puppet Name: puppetdb_metrics_tidy
30 0 * * * /opt/puppetlabs/puppet-metrics-collector/scripts/puppetdb_metrics_tidy
# Puppet Name: orchestrator_metrics_collection
*/5 * * * * /opt/puppetlabs/puppet-metrics-collector/scripts/tk_metrics --metrics_type orchestrator --output-dir /opt/puppetlabs/puppet-metrics-collector/orchestrator --no-print
# Puppet Name: orchestrator_metrics_tidy
0 0 * * * /opt/puppetlabs/puppet-metrics-collector/scripts/orchestrator_metrics_tidy
```

**Comment by tkishel** [comment]
Should be addressed by #47

**Comment by tkishel** [comment]
Fixed by 47
